backend/services/shopify_importer.py

```
@@ -0,0 +1,265 @@
+
+    col_date = hdr.get("date") or hdr.get("day") or "Date"
+
+    # common fields in Shopify conversion report
+    col_sessions = None
+    for k, v in hdr.items():
+        if "session" in k:
+            col_sessions = v; break
+
+    col_conv = None
+    for k, v in hdr.items():
+        if "conversion" in k and "%" in v.lower() or "rate" in k:
```

Ruff (0.13.3)

Parenthesize `a and b` expressions when chaining `and` and `or` together, to make the precedence clear

Parenthesize the `and` subexpression

(RUF021)